R/as-ped-cr.R

```diff
@@ -0,0 +1,195 @@
+  ped_sets <- vector(mode = "list", length = length(status))
+  for (i in 1:length(status)) {
+    current_data <- f_data
+    current_data[[status_str]][f_data[[status_str]] != status_num[i]] <- 0L
+    current_data[[status_str]][f_data[[status_str]] == status_num[i]] <- 1L
+    if (!is.data.frame(data) & length(data) == 2L) {
+      current_data <- list(current_data, data[[2L]])
+    }
+    ped_sets[[i]] <- as_ped(data = current_data, formula = formula, 
+                            cut = cut[[i]], ...)
+    if (output != "list") {
+      ped_sets[[i]]$cause <- as.factor(as.character(status[i]))
+    }
+    class(ped_sets[[i]]) <- c("ped", "data.frame")
+  }
+  if (output == "list") {
+    ped <- ped_sets
+    names(ped) <- status
+    class(ped) <- c("ped_cr_list", "ped_cr")
+    attr(ped, "risks") <- status
+    return(ped)
+  } else {
+    ped <- as.data.frame(Reduce("rbind", ped_sets))
+    class(ped) <- c("ped_cr_df", "ped_cr", "data.frame")
+    attr(ped, "intvars") <- c(attr(ped, "intvars"), "cause")
+    attr(ped, "breaks") <- cut
+    attr(ped, "trafo_args")[["cut"]] <- cut
+    attr(ped, "risks") <- status
+    return(ped)
+  }
+}
+
+#' Conversion to numeric risks
+#' 
+#' This function converts character or factor status into numeric ones.
+#' This is necessary for further processing.
+#' @param data A data.frame of class ped_cr that features time-to-event data
+#' and convariates. (see https://adibender.github.io/pammtools/)
+#' @param status_str A character indicating the colname of the status variable.
+#' @param censor_code A value (character/numeric/factor) indicating as what
+#' censoring has been coded.
+#' @return data with (converted) numeric status column
+make_numeric <- function(data, status_str, censor_code) {
+  risks <- unique(data[[status_str]])
+  risks <- risks[risks != censor_code]
+  if (is.numeric(risks)) {
+    #otherwise not intuitive.
+    risks <- risks[order(risks)]
+  }
+  attr(data, "risks") <- risks
+  data[data[[status_str]] == censor_code, status_str] <- 0
+  for (i in 1:length(risks)) {
+    data[data[[status_str]] == risks[i], status_str] <- i
+  }
+  data[[status_str]] <- as.numeric(data[[status_str]])
+  data
+}
+
+check_cuts <- function(cut, status) {
+  if (is.null(cut)) {
+    return(rep(list(cut), length(status)))
+  }
+  if ((!is.list(cut)) & (!is.numeric(cut))) {
+    stop("cut must be either a numeric vector or a list of numeric vectors.")
+  }
+  if (is.list(cut)) {
+    if (any((!unlist(lapply(cut, is.numeric))) & (!unlist(lapply(cut, is.null))))) {
+      stop("cut must be either a numeric vector or a list of numeric vectors.")
+    }
+  }
+  if ((length(cut) != length(status)) & (!is.numeric(cut))) {
+    stop("cut must have length 1 or identical length as the number of status in the data.")
+  }
+  if (is.numeric(cut)) {
+    return(rep(list(cut), length(status)))
+  } else {
+    return(cut)
+  }
+}
+
+check_data <- function(data) {
+  if (!(is.data.frame(data)) & !(is.list(data))) {
```

Comment:
Does this also work with tibbles, data.tables?

Reply:
tibbles yes, not sure for data.tables. Gonna look into that.

Reply:
It does (tests are added)